mcs/class/Mono.Posix/Test/Mono.Unix.Native/SocketTest.cs

```diff
@@ -206,6 +206,375 @@ public void Shutdown ()
+
+			Assert.AreEqual ("{sin_family=AF_INET, sin_port=htons(5678), sin_addr=127.0.0.1}", address1.ToString ());
+		}
+
+		[Test]
+		public void SockaddrIn6Test ()
+		{
+			var address1 = new SockaddrIn6 {
+				sin6_family = UnixAddressFamily.AF_INET6,
+				sin6_port = Syscall.htons (1234),
+				sin6_flowinfo = 2,
+				sin6_addr = NativeConvert.ToIn6Addr (IPAddress.IPv6Loopback),
+				sin6_scope_id = 3
+			};
+
+			var storage = address1.ToSockaddrStorage ();
+			var address2 = SockaddrIn6.FromSockaddrStorage (storage);
+			Assert.AreEqual (address1, address2);
+
+			var sockaddr = Sockaddr.FromSockaddrStorage (storage);
+			Assert.AreEqual (sockaddr.sa_family, address1.sa_family);
+
+			Assert.AreEqual ("{sin6_family=AF_INET6, sin6_port=htons(1234), sin6_flowinfo=2, sin6_addr=::1, sin6_scope_id=3}", address1.ToString ());
+		}
+
+		[Test]
+		public void BindConnect ()
+		{
+			WithSockets (UnixAddressFamily.AF_INET, UnixSocketType.SOCK_DGRAM, UnixSocketProtocol.IPPROTO_UDP, (so1, so2) => {
+				// Bind UDP socket so1 to 127.0.0.1 with dynamic port
+				var address = new SockaddrIn {
+					sin_family = UnixAddressFamily.AF_INET,
+					sin_port = Syscall.htons (0),
+					sin_addr = new InAddr (127, 0, 0, 1),
+				};
+				if (Syscall.bind (so1, address) < 0)
+					UnixMarshal.ThrowExceptionForLastError ();
+
+				// Get actual port number using getsockname()
+				var actualAddress = new SockaddrIn ();
+				if (Syscall.getsockname (so1, actualAddress) < 0)
+					UnixMarshal.ThrowExceptionForLastError ();
+				Assert.AreEqual (actualAddress.sa_family, UnixAddressFamily.AF_INET);
+				var port = Syscall.ntohs (actualAddress.sin_port);
+				Assert.IsTrue (port != 0);
+
+
+				// Connect so2 to so1
+				var remoteAddress = new SockaddrIn {
+					sin_family = UnixAddressFamily.AF_INET,
+					sin_port = Syscall.htons (port),
+					sin_addr = new InAddr (127, 0, 0, 1),
+				};
+				if (Syscall.connect (so2, remoteAddress) < 0)
+					UnixMarshal.ThrowExceptionForLastError ();
+
+				// Verify peer address using getpeername()
+				var address2 = new SockaddrIn ();
+				if (Syscall.getpeername (so2, address2) < 0)
+					UnixMarshal.ThrowExceptionForLastError ();
+				Assert.AreEqual (address2.sa_family, UnixAddressFamily.AF_INET);
+				Assert.AreEqual (remoteAddress.sin_port, address2.sin_port);
+				Assert.AreEqual (remoteAddress.sin_addr, address2.sin_addr);
+
+				// Send and receive a few bytes
+				long ret;
+				var buffer1 = new byte[] { 42, 43, 44 };
+				ret = Syscall.send (so2, buffer1, (ulong) buffer1.Length, 0);
+				if (ret < 0)
+					UnixMarshal.ThrowExceptionForLastError ();
+
+				var buffer2 = new byte[1024];
+				ret = Syscall.recv (so1, buffer2, (ulong) buffer2.Length, 0);
+				if (ret < 0)
+					UnixMarshal.ThrowExceptionForLastError ();
+
+				Assert.AreEqual (buffer1.Length, ret);
+				for (int i = 0; i < buffer1.Length; i++)
+					Assert.AreEqual (buffer1[i], buffer2[i]);
+			});
+		}
+
+		[Test]
+		public void IPv6 ()
```

Comment:
This test is failing on some of the Jenkins machines:

	Test Case Failures:
	1) MonoTests.Mono.Unix.Native.SocketTest.IPv6 : Mono.Unix.UnixIOException : Cannot assign requested address [EADDRNOTAVAIL].
	at Mono.Unix.UnixMarshal.ThrowExceptionForLastError () [0x00000] in /var/lib/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/Mono.Posix/Mono.Unix/UnixMarshal.cs:460
	at MonoTests.Mono.Unix.Native.SocketTest+<IPv6>c__AnonStorey0.<>m__0 (Int32 so1, Int32 so2) [0x00012] in /var/lib/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/Mono.Posix/Test/Mono.Unix.Native/SocketTest.cs:405
	at MonoTests.Mono.Unix.Native.SocketTest.WithSockets (UnixAddressFamily af, UnixSocketType type, UnixSocketProtocol protocol, System.Action`2 f) [0x00038] in /var/lib/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/Mono.Posix/Test/Mono.Unix.Native/SocketTest.cs:89
	at MonoTests.Mono.Unix.Native.SocketTest.IPv6 () [0x00037] in /var/lib/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/Mono.Posix/Test/Mono.Unix.Native/SocketTest.cs:403
	at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
	at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00038] in /var/lib/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/corlib/System.Reflection/MonoMethod.cs:295

The problem is that not all machines have IPv6 support.

Could you please add a check for `Socket.OSSupportsIPv6` and `Assert.Ignore()` this (and other?) tests if the local machine doesn't support IPv6?